src/main/java/com/mongodb/hibernate/internal/translate/mongoast/filter/AstComparisonFilterOperation.java

```diff
@@ -19,8 +19,16 @@
 import com.mongodb.hibernate.internal.translate.mongoast.AstValue;
 import org.bson.BsonWriter;
 
-public record AstComparisonFilterOperation(AstComparisonFilterOperator operator, AstValue value)
-        implements AstFilterOperation {
+public class AstComparisonFilterOperation implements AstFilterOperation {
```

Comment:
```suggestion
public final class AstComparisonFilterOperation implements AstFilterOperation {
```

Reply:
Now I'm curious why we're adding new AST types that are classes, when all the previous ones were records.

Reply:
I missed that. This class indeed should be a `record`. As far as I can see, it's the only situation where we wrongly implemented `AstNode` as a `class`.

Reply:
I overlooked. Changed to record usage.